src/shared/syscheck_op.c

```diff
@@ -1089,12 +1099,20 @@ char *get_registry_group(char **sid, HANDLE hndl) {
         DWORD dwErrorCode = 0;
 
         dwErrorCode = GetLastError();
-
         if (dwErrorCode == ERROR_NONE_MAPPED) {
             mdebug1("Group not found for registry key");
         }
         else {
-            merror("Error in LookupAccountSid.");
+            LPSTR messageBuffer = NULL;
+            LPSTR end;
+
+            FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                          NULL, dwErrorCode, 0, (LPTSTR) &messageBuffer, 0, NULL);
+            if (end = strchr(messageBuffer, '\r'), end) {
+                *end = '\0';
+            }
+
+            mwarn(FIM_REGISTRY_ACC_SID, "group", dwErrorCode, messageBuffer);
```

Comment:
You are not freeing the memory allocated in messageBuffer!

Reply:
Done!